crates/dojo-core/src/world.cairo

```diff
@@ -356,27 +374,41 @@ mod World {
     fn set_executor(contract_address: ContractAddress) {
         // Only Admin can set executor
         assert(is_account_admin(), 'only admin can set executor');
-        executor_dispatcher::write(IExecutorDispatcher {
-            contract_address: contract_address
-        });
+        executor_dispatcher::write(IExecutorDispatcher { contract_address: contract_address });
     }
 
     #[view]
     fn executor() -> ContractAddress {
         executor_dispatcher::read().contract_address
     }
 
-    /// Set the execution role to be assumed
-    ///
+    /// Validate that the role to be assumed has the required permissions
+    /// then set the execution role
+    /// 
     /// # Arguments
     ///
     /// * `role_id` - The role id to be assumed
+    /// * `components` - The components to be validated
     #[external]
-    fn set_execution_role(role_id: u250) {
+    fn assume_role(role_id: u250, components: Array<ShortString>) {
```

Comment:
Would it be better to use systems here? Is there a way to get all the components that would be written?

Reply:
Hmm how would that work? It is a bit clunky but we can abstract it with the sdk. I'm not sure if there is a more efficient way to do this

Reply:
I was thinking to use the [dependencies](https://github.com/dojoengine/dojo/blob/main/crates/dojo-lang/src/system.rs#L71) to get the components directly that the systems are writing to. If this can be abstracted with the sdk, that could be cool as it's a bit hard to know all components that will be modified